logstash-core/build.gradle

```diff
@@ -0,0 +1,76 @@
+import java.nio.file.Files
+import static java.nio.file.StandardCopyOption.REPLACE_EXISTING
+
+apply plugin: 'java'
+apply plugin: 'idea'
+
+group = 'org.logstash'
+version = '5.0.0-alpha6'
+
+description = """Logstash Core Java"""
+
+sourceCompatibility = 1.8
+targetCompatibility = 1.8
+
+repositories {
+  mavenCentral()
+}
+
+configurations {
+    provided
+}
+
+project.sourceSets {
+    main.compileClasspath += project.configurations.provided
+    main.runtimeClasspath += project.configurations.provided
+    test.compileClasspath += project.configurations.provided
+    test.runtimeClasspath += project.configurations.provided
+}
+project.javadoc.classpath += project.configurations.provided
+
+idea {
+    module {
+        scopes.PROVIDED.plus += [project.configurations.provided]
+    }
+}
+
+
+dependencies {
+  runtime 'org.apache.logging.log4j:log4j-1.2-api:2.6.2'
```

Comment:
Here's my opinion about it :
- in logstash, logging system and configuration is common and shared with everybody : core code and plugins (contrary to an applications server for example)
- manage this shared logging system and its integration is logstash's responsibility, and not each plugin's responsibility.
- if it's not done, some plugin logs will be lost, some others will arrive anywhere.
- in particular for slf4j, it's important that slf4j "api" and "impl" versions are aligned. And logstash-core is the best place to ensure that.

=> To me, it is relevant to put these jars here.

@ph @suyograo @jordansissel 
Any opinion on that subject ?